Builders/WindowsBuilder.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.IO;
+using osu.Desktop.Deploy.Uploaders;
+
+namespace osu.Desktop.Deploy.Builders
+{
+    public class WindowsBuilder : Builder
+    {
+        private const string app_name = "osu!.exe";
+        private const string os_name = "win";
+        private const string channel = "win";
+
+        private readonly string? codeSigningPassword;
+
+        public WindowsBuilder(string version, string? codeSigningPassword)
+            : base(version)
+        {
+            if (!string.IsNullOrEmpty(Program.WindowsCodeSigningCertPath))
+                this.codeSigningPassword = codeSigningPassword ?? Program.ReadLineMasked("Enter code signing password: ");
+        }
+
+        protected override string TargetFramework => "net8.0";
+        protected override string RuntimeIdentifier => $"{os_name}-x64";
+
+        public override Uploader CreateUploader()
+        {
+            string installIcon = Path.Combine(Environment.CurrentDirectory, "install.ico");
+            string extraArgs = $" --splashImage=\"{SplashImagePath}\""
+                               + $" --icon=\"{installIcon}\""
+                               + $" --noPortable";
+
+            if (!string.IsNullOrEmpty(Program.WindowsCodeSigningCertPath))
+                extraArgs += $" --signParams=\"/td sha256 /fd sha256 /f {Program.WindowsCodeSigningCertPath} /p {codeSigningPassword} /tr http://timestamp.comodoca.com\"";
+
+            return new WindowsVelopackUploader(app_name, os_name, RuntimeIdentifier, channel, extraArgs: extraArgs);
+        }
+
+        public override void Build()
+        {
+            RunDotnetPublish();
+
+            bool rcEditCommand =
+                Program.RunCommand("tools/rcedit-x64.exe", $"\"{Path.Combine(Program.StagingPath, "osu!.exe")}\""
+                                                           + $" --set-icon \"{IconPath}\"",
+                    exitOnFail: false);
+
+            if (!rcEditCommand)
+            {
+                // Retry again with wine
+                // TODO: Should probably change this to use RuntimeInfo.OS checks instead of fail values
+                bool wineRcEditCommand =
+                    Program.RunCommand("wine", $"\"{Path.GetFullPath("tools/rcedit-x64.exe")}\""
+                                               + $" \"{Path.Combine(Program.StagingPath, "osu!.exe")}\""
+                                               + $" --set-icon \"{IconPath}\"",
+                        exitOnFail: false);
```

Comment:
This is how it is in `master`, dating back to 2018 or so. I don't remember the reason why this exists in the first place, but It looks like .NET is handling the app icon fine via [`ApplicationIcon`](https://github.com/ppy/osu/blob/1017a124ab9afd5512b7e9f5a05d18954be70579/osu.Desktop/osu.Desktop.csproj#L10) at least when compiled on Windows (which is what our runners do anyway). So... we can maybe remove this, but as I mentioned above I intended to keep the code familiar and not introduce too many changes.

Reply:
Oh okay. I checked and it looked like it didn't exist but I guess I missed it.

Reply:
Maybe I'm going crazy. I definitely thought I saw it there but I can't find it, and it looks like it was added in this PR (the Wine part)...

I guess it's not a big issue either way? Though I still want to explore removing it altogether.

Reply:
maybe @smallketchup82 can answer this then

Reply:
Documenting here for now because I was about to remove it... It appears that `ApplicationIcon` sets the icon correctly for the shortcut, taskbar, and alt-tab, but the window titlebar loses its icon.

Which is weird because we have SDL code to handle that... https://github.com/ppy/osu-framework/blob/e0e945672d41c395c1977d273220cb17f9d52e2a/osu.Framework/Platform/Windows/SDL2WindowsWindow.cs#L293

Needs more investigation.

Reply:
> maybe @smallketchup82 can answer this then

Yes I added this in. Meant to be a sort of "fallback" method when compiling for windows on a linux system. There's no equivalent unix binary for that tool, so the best way to run it is via wine.

Seems to work perfectly, and removing it more or less breaks cross compilation. Best to keep it in

Reply:
Velopack now has a cross-platform resource editor written in C#, as it needs to set icons on it's own Update.exe even when on Linux. I guess you could steal this and make your own small utility instead of using wine+rcedit